src/bp/lang-server/service/index.ts

```diff
@@ -0,0 +1,226 @@
+    const models: Dic<ModelSet> = {}
+    const _scopedAddPairModelToModels = this._addPairModelToModels(models)
+
+    _.chain(files)
+      .map(this._getModelInfoFromFile)
+      .reject(_.isEmpty)
+      .groupBy(model => [model.domain, model.langCode])
+      .forEach(_scopedAddPairModelToModels)
+      .value()
+
+    return models
+  }
+
+  private _getPairModels(lang: string, fastTextModelName: string, bpeModelName: string) {
+    const fastTextModelPath = path.join(this.langDir, fastTextModelName)
+    const bpeModelPath = path.join(this.langDir, bpeModelName)
+
+    try {
+      if (!fs.existsSync(fastTextModelPath)) {
+        throw new Error(`Model '${lang}' at location '${fastTextModelPath}' does not exist`)
+      }
+      if (!fs.existsSync(bpeModelPath)) {
+        throw new Error(`Model '${lang}' at location '${bpeModelPath}' does not exist`)
+      }
+    } catch (err) {
+      throw new VError(`Could not find model '${lang}' in '${this.langDir}'`, err)
+    }
+
+    const fastTextModel = {
+      name: lang,
+      path: fastTextModelPath,
+      sizeInMb: 0,
+      loaded: false
+    }
+
+    const bpeModel = {
+      name: lang,
+      path: bpeModelPath,
+      sizeInMb: 0,
+      loaded: false
+    }
+
+    return { fastTextModel, bpeModel }
+  }
+
+  @WrapErrorsWith(args => `Couldn't load language model "${args[0]}"`)
+  private async _loadModels(lang: string) {
+    const fastTextModel = await this._loadFastTextModel(lang)
+    const bpeModel = await this._loadBPEModel(lang)
+    this._models[lang] = { fastTextModel, bpeModel }
+  }
+
+  private async _loadFastTextModel(lang: string): Promise<LoadedFastTextModel> {
+    const loadingAction = async (lang: string) => {
+      const model = new toolkit.FastText.Model(false, true, true)
+      const path = this._models[lang].fastTextModel.path
+      await model.loadFromFile(path)
+      return { model, path }
+    }
+
+    const { model, usedDelta } = await this._getRessourceConsumptionInfo(lang, loadingAction)
+
+    return {
+      ...this._models[lang].fastTextModel,
+      model,
+      sizeInMb: usedDelta,
+      loaded: true
+    } as LoadedFastTextModel
+  }
+
+  private async _loadBPEModel(lang: string): Promise<LoadedBPEModel> {
+    const loadingAction = lang => {
+      const tokenizer = toolkit.SentencePiece.createProcessor()
+      const path = this._models[lang].bpeModel.path
+      tokenizer.loadModel(path)
+      return Promise.resolve({ model: tokenizer, path })
+    }
+
+    const { model: tokenizer, usedDelta } = await this._getRessourceConsumptionInfo(lang, loadingAction)
+
+    return {
+      ...this._models[lang].bpeModel,
+      tokenizer,
+      sizeInMb: usedDelta,
+      loaded: true
+    } as LoadedBPEModel
+  }
+
+  private async _getRessourceConsumptionInfo(
+    lang: string,
+    action: (
+      lang: string
+    ) => Promise<{ model: MLToolkit.SentencePiece.Processor | MLToolkit.FastText.Model; path: string }>
```

Comment:
Maybe a personal preference, but when you have declarations to do, create an interface for them, even if there's only two properties. It makes it a lot easier to read the actual code